app/controllers/chat_channels_controller.rb

```diff
@@ -112,6 +112,13 @@ def open_chat
     end
   end
 
+  def block_chat
+    chat_channel = ChatChannel.find(params[:chat_id])
```

Comment:
This method needs to authorize that the `current_user` is permitted to take this action—e.g. that the channel is direct and that they belong to it. This should be done through the appropriate chat channel policy.

Reply:
Added policy after 116 and wrapped status change in if statement that validates the channel, checks that it is direct, and if the user belongs to it.